[AIRFLOW-1384] Add to README.md CaDC/ARGO

vr00n opened the pull request:
added to  currently **officially** using Airflow:

[California Data Collaborative](http://californiadatacollaborative.org) powered by [ARGO Labs](http://www.argolabs.org)

Dear Airflow maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!


### JIRA
- [x] My PR addresses the following [Airflow JIRA] **https://issues.apache.org/jira/browse/AIRFLOW-1384**


### Description
- The California Data Collaborative is a unique coalition of forward thinking municipal water managers in California who along with ARGO, a startup non-profit that builds, operates, and maintains data infrastructures, are pioneering new standards of collaborating around and administering water data for millions Californians. 

ARGO has deployed a hosted version of Airflow on AWS and it is used to orchestrate data pipelines to parse water use data from participating utilities to power analytics. Furthermore, ARGO also uses Airflow to power a data infrastructure for citywide street maintenance via https://github.com/ARGO-SQUID


### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
Change to README.md does not require unit testing.

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

vr00n committed:
Update README.md

added to  currently **officially** using Airflow section of README.md

[California Data Collaborative](https://github.com/California-Data-Collaborative) powered by [ARGO Labs](http://www.argolabs.org)

christophertull reviewed (approved):
I approve of this change.

vr00n committed:
Added CaDC/ARGO Labs to README.md

Please consider adding [Argo Labs](www.argolabs.org) to the Airflow users section.
**Context**
- The California Data Collaborative is a unique coalition of forward thinking municipal water managers in California who along with ARGO, a startup non-profit that builds, operates, and maintains data infrastructures, are pioneering new standards of collaborating around and administering water data for millions Californians. 

- ARGO has deployed a hosted version of Airflow on AWS and it is used to orchestrate data pipelines to parse water use data from participating utilities to power analytics. Furthermore, ARGO also uses Airflow to power a data infrastructure for citywide street maintenance via https://github.com/ARGO-SQUID
